working/2364 - primary constructors/feature-specification.md

```diff
@@ -233,6 +236,14 @@ class Point {
 class Point(int x, [int y = 0]);
 ```
 
+We can omit the type of an optional parameter with a default value,
+in which case the type is inferred from the default value:
+
+```dart
+// Infer the type of `y` from the default value.
+class Point(int x, [y = 0]);
```

Comment:
Do we support this inference only for primary constructor parameters? This seems like it would be useful for all parameters (and I expect users would intuit that if it works for parameters in one place, it would work for them everywhere).

Reply:
Right, https://github.com/dart-lang/language/issues/3970.